src/main/java/com/onarandombox/MultiverseCore/utils/PermissionsTool.java

```diff
@@ -0,0 +1,329 @@
+
+        setUpWorldPermissionWildcards();
+    }
+
+    /**
+     * Registers the permissions for a destination.
+     *
+     * @param destination The destination to register permissions for.
+     */
+    public void registerDestinationTeleportPermissions(Destination<?> destination) {
+        try {
+            pluginManager.addPermission(new Permission(TELEPORT_PERM_PREFIX + "self." + destination.getIdentifier(), PermissionDefault.OP));
+            pluginManager.addPermission(new Permission(TELEPORT_PERM_PREFIX + "other." + destination.getIdentifier(), PermissionDefault.OP));
+            pluginManager.addPermission(new Permission(TELEPORT_PERM_PREFIX + "self." + destination.getIdentifier() + ".*", PermissionDefault.OP));
+            pluginManager.addPermission(new Permission(TELEPORT_PERM_PREFIX + "other." + destination.getIdentifier() + ".*", PermissionDefault.OP));
+        } catch (IllegalArgumentException e) {
+            Logging.finer("Destination permissions already registered for '" + destination.getIdentifier() + "'");
+        }
+        Logging.finer("Registered permissions for '" + destination.getIdentifier() + "'");
+    }
+
+    /**
+     * Registers the finer permissions for a destination.
+     *
+     * @param destination   The destination to register permissions for.
+     * @param finerSuffix   The finer suffix to register.
+     */
+    public void registerFinerDestinationTeleportPermissions(Destination<?> destination, String finerSuffix) {
+        String finerPermissionName = TELEPORT_PERM_PREFIX + "self." + destination.getIdentifier() + "." + finerSuffix;
+        if (pluginManager.getPermission(finerPermissionName) != null) {
+            return;
+        }
+        Permission permission = pluginManager.getPermission(TELEPORT_PERM_PREFIX + "self." + destination.getIdentifier() + ".*");
+        if (permission == null) {
+            return;
+        }
+
+        Permission finerPermission = new Permission(finerPermissionName, PermissionDefault.OP);
+        try {
+            pluginManager.addPermission(finerPermission);
+            return;
+        } catch (IllegalArgumentException e) {
+            Logging.finer("Finer destination permissions already registered for '" + destination.getIdentifier() + "'");
+        }
+
+        finerPermission.addParent(permission, true);
+        Logging.finer("Registered finer permissions '" + finerPermissionName + "' for '" + destination.getIdentifier() + "'");
+    }
+
+    /**
+     * Checks if a player has the permission to bypass the player limit.
+     *
+     * @param sender    The player to check.
+     * @param toWorld   The world to check.
+     * @return True if the player has the permission, false otherwise.
+     */
+    public boolean hasBypassPlayerLimit(@NotNull CommandSender sender, @NotNull MVWorld toWorld) {
+        return hasPermission(sender, "mv.bypass.playerlimit." + toWorld.getName());
+    }
```

Comment:
I think we should probably split permission checking into a separate class from the class doing permission registration. I wonder, could all the registration stuff be package-private so it's not part of our exported API? This this checking stuff could be in PermissionChecker or something?

Reply:
yea can

Reply:
hm but i realised `package-private` wont work tho, bcu its in the `utils` package, and it will need to be access by other parent packages